nautobot/extras/models/customfields.py

```diff
@@ -325,8 +328,97 @@ def validate(self, value):
+    """
+
+    field = models.ForeignKey(
+        to="extras.CustomField",
+        on_delete=models.CASCADE,
+        related_name="choices",
+        limit_choices_to={"type": CustomFieldTypeChoices.TYPE_SELECT},
+    )
+    value = models.CharField(max_length=100)
+    weight = models.PositiveSmallIntegerField(default=100, help_text="Higher weights appear later in the list")
+
+    class Meta:
+        ordering = ["field", "weight", "value"]
+        unique_together = ["field", "value"]
+
+    def __str__(self):
+        return self.value
+
+    def clean(self):
+        if self.field.type not in (CustomFieldTypeChoices.TYPE_SELECT, CustomFieldTypeChoices.TYPE_MULTISELECT):
+            raise ValidationError("Custom field choices can only be assigned to selection fields.")
+
+    def save(self, *args, **kwargs):
+        """
+        When a custom field choice is saved, perform logic that will update data across all custom field data.
+        """
+        if self.present_in_database:
+            database_object = self.__class__.objects.get(pk=self.pk)
+        else:
+            database_object = self
+
+        super().save(*args, **kwargs)
+
+        if self.value != database_object.value:
+            update_custom_field_choice_data.delay(self.field.pk, database_object.value, self.value)
+
+    def delete(self, *args, **kwargs):
+        """
+        When a custom field choice is deleted, remove references to in custom field data
+        """
+        if self.field.default:
+            # Cannot delete the choice if it is the default value.
+            if self.field.type == CustomFieldTypeChoices.TYPE_SELECT and self.field.default == self.value:
+                raise models.ProtectedError(
+                    self, "Cannot delete this choice because it is the default value for the field."
+                )
+            elif self.value in self.field.default:
+                raise models.ProtectedError(
+                    self, "Cannot delete this choice because it is one of the default values for the field."
+                )
+
+        if self.field.type == CustomFieldTypeChoices.TYPE_SELECT:
+            # Check if this value is in active use in a select field
+            for ct in self.field.content_types.all():
+                model = ct.model_class()
+                if model.objects.filter(**{f"custom_field_data__{self.field.name}": self.value}).exists():
+                    raise models.ProtectedError(self, "Cannot delete this choice because it is in active use.")
+
+        else:
+            # Check if this value is in active use in a multi-select field
+            for ct in self.field.content_types.all():
+                model = ct.model_class()
+                if model.objects.filter(**{f"custom_field_data__{self.field.name}__contains": self.value}).exists():
+                    raise models.ProtectedError(self, "Cannot delete this choice because it is in active use.")
```

Comment:
Same concerns here re: `delete()` overload vs. using signals.

Reply:
In this case, we are actually implementing integrity checks that _will_ veto a delete. To me, that calls for a more direct code path. Because of the UX flow here, we have to explicitly call `delete()` within the formset save logic and catch any exceptions. Thus, moving this logic to a signal does nothing but obscure that code flow IMO.